service/src/main/java/bio/terra/workspace/app/controller/ReferencedGcpResourceController.java

```diff
@@ -268,49 +237,24 @@ public ResponseEntity<ApiGcpGcsBucketResource> getBucketReferenceByName(UUID uui
   @Traced
   @Override
   public ResponseEntity<ApiGcpGcsBucketResource> updateBucketReferenceResource(
-      UUID workspaceUuid, UUID referenceId, ApiUpdateGcsBucketReferenceRequestBody body) {
+      UUID workspaceUuid, UUID resourceId, ApiUpdateGcsBucketReferenceRequestBody body) {
     AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
     workspaceService.validateWorkspaceAndAction(
         userRequest, workspaceUuid, SamWorkspaceAction.UPDATE_REFERENCE);
-    String bucketName = body.getBucketName();
-    CloningInstructions cloningInstructions =
-        CloningInstructions.fromApiModel(body.getCloningInstructions());
-    if (StringUtils.isEmpty(bucketName)) {
-      referenceResourceService.updateReferenceResource(
-          workspaceUuid,
-          referenceId,
-          body.getName(),
-          body.getDescription(),
-          null,
-          cloningInstructions,
-          userRequest);
-    } else {
-      ReferencedGcsBucketResource referencedResource =
-          referenceResourceService
-              .getReferenceResource(workspaceUuid, referenceId)
-              .castByEnum(WsmResourceType.REFERENCED_GCP_GCS_BUCKET);
-      ReferencedGcsBucketResource.Builder updateBucketResourceBuilder =
-          referencedResource.toBuilder().bucketName(bucketName);
-      if (cloningInstructions != null) {
-        // only overwrite if non-null
-        updateBucketResourceBuilder.wsmResourceFields(
-            referencedResource.getWsmResourceFields().toBuilder()
-                .cloningInstructions(cloningInstructions)
-                .build());
-      }
-      referenceResourceService.updateReferenceResource(
-          workspaceUuid,
-          referenceId,
-          body.getName(),
-          body.getDescription(),
-          updateBucketResourceBuilder.build(),
-          null, // passed in via resource argument
-          userRequest);
-    }
-
+    ReferencedResource resource =
+        referencedResourceService.validateReferencedResourceAndAction(
+            userRequest, workspaceUuid, resourceId, SamWorkspaceAction.UPDATE_REFERENCE);
+    wsmResourceService.updateResource(
+        userRequest,
+        resource,
+        new CommonUpdateParameters()
+            .setName(body.getName())
+            .setDescription(body.getDescription())
+            .setCloningInstructions(StewardshipType.REFERENCED, body.getCloningInstructions()),
```

Comment:
Is there a reason why a null check is needed for cloning instructions in the controlled resource controller, but not here in the referenced resource controller?

Is this because for referenced resources, the update fields (aside from name + desc.) are "flattened" (i.e., not nested in an object), whereas controlled resources have an explicit `updateParameters` object in the request body?

Reply:
Yes, since there is no optional parameter set, there is no null pointer to test. The value itself can be null in either case.